store/src/java/com/zimbra/cs/gal/GalSyncToken.java

```diff
@@ -208,8 +208,17 @@ public boolean isEmpty() {
     public void merge(GalSyncToken that) {
         ZimbraLog.gal.debug("merging token "+this+" with "+that);
         mLdapTimestamp = LdapUtil.getEarlierTimestamp(this.mLdapTimestamp, that.mLdapTimestamp);
-        for (String aid : that.mChangeIdMap.keySet())
-            mChangeIdMap.put(aid, that.mChangeIdMap.get(aid));
+        for (String aid : that.mChangeIdMap.keySet()) {
+            String strThatVal = that.mChangeIdMap.get(aid);
+            String strThisVal = this.mChangeIdMap.get(aid);
+            if (StringUtils.isNotBlank(strThatVal) && StringUtils.isNotBlank(strThisVal) &&
+                    StringUtils.isNumeric(strThatVal) && StringUtils.isNumeric(strThisVal)) {
+                int thisVal = Integer.parseInt(this.mChangeIdMap.get(aid));
+                int thatVal = Integer.parseInt(that.mChangeIdMap.get(aid));
+                strThatVal = thisVal > thatVal ? strThisVal : strThatVal;
+            }
+            mChangeIdMap.put(aid, strThatVal);
```

Comment:
Why the preference for value from that (when the if branch isn't executed)?  My intuition would be to take the value from whichever won in getEarlierTimestamp?

Reply:
The thing come into picture when both getEarlierTimestamp is same. but the deleted item in tombstone having higher mod sequence than mail item.

Reply:
Could you add that as a comment please?

Reply:
Done